Examples/ObjectiveC/WebAPIDataExample.m

```diff
@@ -48,48 +48,34 @@ - (void)addItemsToMap:(NSArray *)features {
     // Use MGLCircleStyleLayer to represent the points with simple circles.
     // In this case, we can use style functions to gradually change properties between zoom level 2 and 7: the circle opacity from 50% to 100% and the circle radius from 2pt to 3pt.
     MGLCircleStyleLayer *circles = [[MGLCircleStyleLayer alloc] initWithIdentifier:@"lighthouse-circles" source:source];
-    circles.circleColor = [MGLStyleValue valueWithRawValue:lighthouseColor];
-    circles.circleOpacity = [MGLStyleValue valueWithInterpolationMode:MGLInterpolationModeExponential
-        cameraStops:@{
-            @2: [MGLStyleValue valueWithRawValue:@0.5],
-            @7: [MGLStyleValue valueWithRawValue:@1.0]
-        }
-        options:@{MGLStyleFunctionOptionDefaultValue:[MGLStyleValue valueWithRawValue:@0.75]}];
-                             
-    circles.circleRadius = [MGLStyleValue valueWithInterpolationMode:MGLInterpolationModeInterval
-        cameraStops:@{
-            @2: [MGLStyleValue valueWithRawValue:@2],
-            @7: [MGLStyleValue valueWithRawValue:@3]
-        }
-        options:@{MGLStyleFunctionOptionDefaultValue:@1}];
+    circles.circleColor = [NSExpression expressionForConstantValue:lighthouseColor];
     
+    // The circles should increase in opacity from 0.5 to 1 based on zoom level.
+    circles.circleOpacity = [NSExpression expressionWithFormat:@"mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)",
+                             @{@2: @0.5, @7: @1.0 }];
+    circles.circleRadius = [NSExpression mgl_expressionForSteppingExpression: NSExpression.zoomLevelVariableExpression
+                                                              fromExpression:[NSExpression expressionForConstantValue:@1]
+                                                                       stops:[NSExpression expressionForConstantValue: @{@2: @2, @7: @3}]];
```

Comment:
We should be consistent about using either format strings or convenience initializers for things like interpolation and stepping expressions. For consistency with all the interpolation expressions in this PR, we’d use:

```objc
circles.circleRadius = [NSExpression expressionWithFormat:@"mgl_step:from:stops:($zoomLevel, 1, %@)",
                        @{@2: @2, @7: @3}];
```